java/core/src/main/java/sleeper/core/record/process/status/ProcessRun.java

```diff
@@ -110,6 +111,18 @@ public RecordsProcessedSummary getFinishedSummary() {
         }
     }
 
+    public List<ProcessStatusUpdate> getStatusUpdates() {
+        return statusUpdates;
+    }
+
+    public <T extends ProcessStatusUpdate> Optional<ProcessStatusUpdate> getLastStatusOfType(Class<T> cls) {
+        List<ProcessStatusUpdate> updatesLatestFirst = new ArrayList<>(getStatusUpdates());
+        Collections.reverse(updatesLatestFirst);
```

Comment:
I'm not sure this is actually the order in which the status updates will be held in the list. Can this be tested against JobStatusUpdates.streamFrom to ensure the ordering is correct?

Reply:
In `ProcessRun` the status updates are stored oldest first, In `ProcessRuns.fromRecordsLatestFirst` it needs to find the start of a run, so loops through the latest first list backwards, adding status updates to `ProcessRun` oldest first. `JobStatusUpdates.streamFrom` delegates to `ProcessRuns.fromRecordsLatestFirst`.

I agree that it's a bit confusing having to reverse the list so many times though. Perhaps renaming the getter for statusUpdates to `getStatusUpdatesOldestFirst()`, or adding another getter for latestFirst?

Reply:
Excuse me, I can see now it is testing against `JobStatusUpdates.streamFrom` already. The updates are stored in ProcessRun oldest first because it iterates through the records backwards in `ProcessRuns.fromRecordsLatestFirst`, to find the start of each run. Maybe we can clarify that in ProcessRun?